components/core/DataView.js

```diff
@@ -592,19 +549,18 @@ export default class DataView extends React.Component {
           {header}
           <div css={STYLES_IMAGE_GRID}>
             {this.props.items
-              .slice(this.state.startIndex, this.state.startIndex + VIEW_LIMIT)
+              .slice(this.state.startIndex, this.state.startIndex + this.state.viewLimit)
               .map((each, i) => {
                 const cid = each.ipfs.replace("/ipfs/", "");
                 return (
                   <div
                     key={each.id}
                     css={STYLES_IMAGE_BOX}
-                    onClick={() => this._handleSelect(i + this.state.startIndex)}
+                    onClick={() => this._handleSelect(i)}
                     onMouseEnter={() => this.setState({ hover: i })}
                     onMouseLeave={() => this.setState({ hover: null })}
                   >
                     <SlateMediaObjectPreview
-                      blurhash={each.blurhash}
                       url={`${Constants.gateways.ipfs}/${each.ipfs.replace("/ipfs/", "")}`}
```

Comment:
Can you add `blurhash={each.blurhash}` back into <SlateMediaObjectPreview/>? It looks like it got deleted from the line above in one of your commits. Blurhash is necessary so that the images appear as a gradient blur rather than as a grey box while loading